book/scale_nodes.rst

```diff
@@ -0,0 +1,358 @@
+ORO Application Scaling - Set-up Instructions
+=============================================
+
+Introduction:
+-------------
+
+Slow-down of the server response when application is under heavy loads may result in deterioration of the 
+user-experience and decrease of the user-satisfaction level. Therefore, we have to somehow optimize and speed-up the 
+response time.
+
+The simplest solution is to scale-up web nodes of the application. For example, scaling of just two web nodes can 
+improve the response time twice (at certain amount of request). The article describes how to scale the web-nodes for the 
+application.
+
+Preliminary Recommendations:
+----------------------------
+
+- Share the *[Application root]/app/cache* folder among all the web nodes, for example using NFS.
+
+- Use Nginx (http://nginx.org/) as a proxy balancer
+
+- Use REDIS (http://redis.io/) as a data cache for Annotations, Metadata, ACL, etc.
+
+- Use an attachment storage such as AmazonS3/FTP/SFTP/MongoGridFS/Dropbox or any other supported storage from the 
+  `list <https://github.com/KnpLabs/KnpGaufretteBundle>`_.
+
+  - Use APC as an attachment cache.
```

Comment:
It can be useful for example when we have most of attachments as image attachments.
As we generate thumbnails for such attachments and they are stored in web folder (which is private per each web node) it will be faster to get original image from APC cache to generate thumbnail than getting it from storage.